src/sync-controller.js

```diff
@@ -42,7 +43,12 @@ export const syncPointStrategies = [
 
         if (segment.dateTimeObject) {
           const segmentTime = segment.dateTimeObject.getTime() / 1000;
-          const segmentStart = segmentTime + syncController.datetimeToDisplayTime;
+          let segmentStart = segmentTime + syncController.datetimeToDisplayTime;
+
+          // make sure segment start time is not higher than total segment time
+          if (segment.discontinuity && segmentStart > totalSegmentTime) {
+            segmentStart = totalSegmentTime;
```

Comment:
This will not work for a real live playlist. The point of `segmentStart` here is to know the start time of the segment relative to when the stream was first joined. In a live playlist, segments fall off the window, so starting `totalSegmentTime = 0` is inaccurate as you lose the total segment time that has fallen off the live window. 

What I think should be done is `syncController.datetimeToDisplayTime` should be tracked per discontinuity timeline, instead of once at the start for the whole stream, similar to how the time mapping objects are stored per timeline in `syncController.timelines`

Reply:
Thanks, this is a really good idea!